static/js/containers/SettingsPage.js

```diff
@@ -0,0 +1,81 @@
+/* global SETTINGS: false */
+import React from 'react';
+import { connect } from 'react-redux';
+import Loader from 'react-loader';
+
+import Jumbotron from '../components/Jumbotron';
+import {
+  startProfileEdit,
+  FETCH_PROCESSING,
+  clearProfile,
+} from '../actions/index';
+import ProfileFormContainer from './ProfileFormContainer';
+
+class SettingsPage extends ProfileFormContainer {
+
+  componentDidMount() {
+    this.fetchProfile();
+  }
+
+  componentDidUpdate() {
+    this.fetchProfile();
+  }
+
+  componentWillUnmount() {
+    const { dispatch, params: { username } } = this.props;
```

Comment:
@aliceriot i dont think so, I think this is injecting because of routs. https://github.com/mitodl/micromasters/blob/master/static/js/dashboard_routes.js#L43, 

I checked it, even if it add this line in settings ``export default connect(ProfileFormContainer.mapStateToProps)(UserPage);`` it do not supply user name param